EduVPN/ConfigurationService.swift

```diff
@@ -12,17 +12,52 @@ import AppAuth
 /// Fetches configuration
 class ConfigurationService {
     
-    enum Error: Swift.Error {
+    enum Error: Int, LocalizedError {
         case unknown
-        case notImplemented
         case invalidURL
         case missingToken
         case invalidKeyPair
         case invalidConfiguration
+        
+        var localizedDescription: String {
+            switch self {
+            case .unknown:
+                return NSLocalizedString("Configuration failed for unknown reason", comment: "")
+            case .invalidURL:
+                return NSLocalizedString("Configuration failed because provider info was invalid", comment: "")
+            case .missingToken:
+                return NSLocalizedString("Configuration could not be retrieved because no valid token was available", comment: "")
+            case .invalidKeyPair:
+                return NSLocalizedString("Invalid keypair received from provider", comment: "")
+            case .invalidConfiguration:
+                return NSLocalizedString("Invalid configuration received from provider", comment: "")
+            }
+        }
+        
+        var recoverySuggestion: String? {
+            switch self {
+            case .unknown:
+                return NSLocalizedString("Try to connect again.", comment: "")
+            case .invalidURL:
+                return NSLocalizedString("Go back to the first screen and try again.", comment: "")
+            case .missingToken:
+                return NSLocalizedString("Try to authorize again with your provider.", comment: "")
+            case .invalidKeyPair:
+                return NSLocalizedString("Try to connect again later.", comment: "")
+            case .invalidConfiguration:
+                return NSLocalizedString("Try to connect again later.", comment: "")
+            }
+        }
     }
     
+    /// Fetches configuration for a profile including certificate and private key
+    ///
+    /// - Parameters:
+    ///   - profile: Profile
+    ///   - authState: Authencation token
+    ///   - handler: Config or error
     func configure(for profile: Profile, authState: OIDAuthState, handler: @escaping (Either<Config>) -> ()) {
```

Comment:
Shouldn't this be `Result<Config>`? Traditionally `Either` has two generic parameters (and the two cases `.left` and `.right`), while `Result` only has one generic parameter, since the other is always an `Error`.